### PR TITLE
Adds new dynamic programming algorithm

### PR DESCRIPTION
**Pull request by kishan151999**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+"""
+Longest Palindromic Subsequence Problem:
+Find the longest sequence of characters from a given input string that
+is the same backwards and forwards
+https://en.wikipedia.org/wiki/Longest_common_subsequence_problem
+"""
+
+
+def LongestPalindromicSubsequence(string, n):
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:9]:
```suggestion
def longest_palindromic_subsequence(s: str, n: int) -> str:
```
Python function names are in snake_case as discussed in CONTRIBUTING.md.

Python type hints as discussed in CONTRIBUTING.md.

`string` is the name of a Python standard library module so should not be used as a variable name.

Can we come up with a more self-documenting variable name than `n`?

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            elif string[y] == string[z]:
+                L[y][z] = L[y + 1][z - 1] + 2
+            else:
+                L[y][z] = max(L[y][z - 1], L[y + 1][z])
+
+    return L[0][n - 1]
+
+
+if __name__ == "__main__":
+    testCase = "ABAC"
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:46]:
```suggestion
    test_case = "ABAC"
```

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    >>> LongestPalindromicSubsequence("ABBCDABBC",9)
+    5
+    >>> LongestPalindromicSubsequence("ABACCG",6)
+    3
+    >>> LongestPalindromicSubsequence("55055901565109",14)
+    9
+    """
+
+    # creating an array to store the values generated
+    L = [[1 for i in range(n)] for i in range(n)]
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:29]: Uppercase is reserved for constant literals in Python.

**Comment by cclauss** [comment]
What is the use case for this algorithm?  Where would we want to use it and why?

File: dynamic_programming/longest_palindromic_subsequence.py (hunk 3 above)
Comment (kishan151999) [reply dynamic_programming/longest_palindromic_subsequence.py:29]: Sorry for the mistakes, I will fix the issues.

**Comment by kishan151999** [comment]
> What is the use case for this algorithm? Where would we want to use it and why?
The uses are mostly to do with biology and more specifically DNA. They are used for DNA marking and permitting cutting. They can also be used to change a 1 dimensional DNA chain into a 2 or 3 dimensional structure. It can also be used to compress DNA sequences to save space. 

--https://pubmed.ncbi.nlm.nih.gov/11700586/

--https://stackoverflow.com/questions/23861436/real-life-situations-using-palindrome-algorithm#:~:text=Palindromes%20are%20used%20in%20DNA,2%20or%203%20dimensional%20structure.&text=Palindromes%20are%20strings%20that%20read,plan%2C%20a%20canal%2C%20Panama!

**Comment by kishan151999** [comment]
I have fixed all the issues and committed the changes. Thank you for letting me know.

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+https://en.wikipedia.org/wiki/Longest_common_subsequence_problem
+"""
+
+
+def longest_palindromic_subsequence(s: str, length: int) -> int:
+
+    """
+    Computes the longest Palindromic sequence of a string
+
+    :param s: str, the string we provide
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:14]: This comment can be removed because it tells us nothing that is not already in the call signature.

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    :param length: int, the length of the string
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:15]: We can calculate the length of the string with `len(s)` so why require the caller to provide it?

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    >>> longest_palindromic_subsequence("ABBCDABBC",9)
+    5
+    >>> longest_palindromic_subsequence("ABACCG",6)
+    3
+    >>> longest_palindromic_subsequence("55055901565109",14)
+    9
+    """
+
+    # creating an array to store the values generated
+    dp_table = [[1 for i in range(length)] for i in range(length)]
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:29]: Can we come up with a more self-documenting name than `dp_table`?

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
+"""
+Longest Palindromic Subsequence Problem:
+Find the longest sequence of characters from a given input string that
+is the same backwards and forwards
+https://en.wikipedia.org/wiki/Longest_common_subsequence_problem
+"""
+
+
+def longest_palindromic_subsequence(s: str, length: int) -> int:
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:9]:
```suggestion
def longest_palindromic_subsequence(s: str, length: int) -> str:
```
We should return the sequence, not the length of the sequence.  The function is not named `length_of_the_longest_palindromic_subsequence()` so I would not expect an int result.

The caller can always call `len(result)` if they want the length.

File: dynamic_programming/longest_palindromic_subsequence.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    :return L[0][n - 1]: int, the length of the longest Palindromic subsequence
+
+
+
+    >>> longest_palindromic_subsequence("ABBCDABBC",9)
+    5
+    >>> longest_palindromic_subsequence("ABACCG",6)
+    3
+    >>> longest_palindromic_subsequence("55055901565109",14)
+    9
```
Comment (cclauss) [inline dynamic_programming/longest_palindromic_subsequence.py:25]: Please provide a negative test case that contains no palendrome.  Also, a test case for the empty string.

File: dynamic_programming/longest_palindromic_subsequence.py (hunk 6 above)
Comment (kishan151999) [reply dynamic_programming/longest_palindromic_subsequence.py:29]: Would array be fine? If not what would you suggest?

Comment (cclauss) [reply dynamic_programming/longest_palindromic_subsequence.py:29]: What would we call the value that is in each cell of this table?  Are the _weights_ or _probabilities_ or _angstroms_ or what?
